### PR TITLE
Add gulp

### LINKED ISSUE #16: Add Gulp to vagrant
Gulp/Elixir is not currently installing as part of the vagrant image.  This fails to install due to the shared folder/symlink issue with vagrant.

It works inside the VM if the vagrant folder is copied to new directory and the installation happens in there.

Steps to install:

As root:

```
apt-get install curl
curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
apt-get install -y nodejs
npm install --global gulp-cli
```

As vagrant user:

```
cd /path-to-files
npm install
```

### PR DESCRIPTION
**Pull request by geeksareforlife**
This adds functionality for gulp, but the windows host script is not tested

Closes #16

### CODE REVIEW

**Comment by geeksareforlife** [comment]
To test:

- download this branch (add-gulp)
- run "vagrant up"
- then try and run "rungulp.cmd"

```
[1:10]  
You should get something like:

[1:11]  
 ```[11:59:25] Using gulpfile /srv/vagrant/gulpfile.js
[11:59:25] Starting 'all'...
[11:59:25] Starting 'sass'...
[11:59:26] Finished 'sass' after 864 ms
[11:59:26] Starting 'webpack'...
[11:59:29] 
[11:59:29] Finished 'webpack' after 3.23 s
[11:59:29] Finished 'all' after 4.1 s
[11:59:29] Starting 'default'...
┌───────────────┬───────────────────────────────┬────────────────────────────────┬────────────────────┐
│ Task          │ Summary                       │ Source Files                   │ Destination        │
├───────────────┼───────────────────────────────┼────────────────────────────────┼────────────────────┤
│ mix.sass()    │ 1. Compiling Sass             │ resources/assets/sass/app.scss │ public/css/app.css │
│               │ 2. Autoprefixing CSS          │                                │                    │
│               │ 3. Concatenating Files        │                                │                    │
│               │ 4. Writing Source Maps        │                                │                    │
│               │ 5. Saving to Destination      │                                │                    │
├───────────────┼───────────────────────────────┼────────────────────────────────┼────────────────────┤
│ mix.webpack() │ 1. Transforming ES2015 to ES5 │ resources/assets/js/app.js     │ public/js/app.js   │
│               │ 2. Writing Source Maps        │                                │                    │
│               │ 3. Saving to Destination      │                                │                    │
└───────────────┴───────────────────────────────┴────────────────────────────────┴────────────────────┘
[11:59:29] Finished 'default' after 10 ms
```

**Comment by geeksareforlife** [comment]
I've merged the laravel update into here and tested it as working

File: dev/vagrant-config/laravel/dogulp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (cooperaj) [inline dev/vagrant-config/laravel/dogulp]: My comment seemed to have disappeared :/

Why are we copying files around to build them? That seems like it'll be introducing a whole bunch of fragility to the process.

**Comment by geeksareforlife** [comment]
gulp won't install on the shared folder within the VM

Weeks of work, encapsulated in one sentence :)

**Pull request merged**